client/tests/webdriver/specs/customFields.spec.js

```diff
@@ -0,0 +1,274 @@
+  describe("For report's custom fields", () => {
+    it("Should open a create new report page", () => {
+      CreateReport.openAsAdminUser()
+      CreateReport.form.waitForExist()
+      CreateReport.form.waitForDisplayed()
+    })
+
+    it("Should not show default invisible fields", () => {
+      // both of these fields invisible at first
+      CreateReport.fieldsToggledVisibilityByTrainButton.forEach(invisField => {
+        expect(invisField.isExisting()).to.equal(false)
+      })
+
+      CreateReport.fieldsNotToggledVisibilityByTrainButton.forEach(
+        invisField => {
+          expect(invisField.isExisting()).to.equal(false)
+        }
+      )
+    })
+
+    it("Should toggle the correct invisible fields", () => {
+      const button = CreateReport.getEngagementTypesButtonByName(
+        TESTED_ENGAGEMENT_BUTTON
+      )
+      button.waitForExist()
+      button.waitForDisplayed()
+      button.click()
+      CreateReport.fieldsToggledVisibilityByTrainButton.forEach(
+        nowVisisbleField => {
+          nowVisisbleField.waitForExist()
+          nowVisisbleField.waitForDisplayed()
+        }
+      )
+
+      CreateReport.fieldsNotToggledVisibilityByTrainButton.forEach(
+        stillInvisField => {
+          expect(stillInvisField.isExisting()).to.equal(false)
+        }
+      )
+    })
+
+    it("Should persist previous valid data when toggling field's visibility", () => {
+      CreateReport.numberTrainedField.setValue(validNumberInput)
+      const button = CreateReport.getEngagementTypesButtonByName(
+        TESTED_ENGAGEMENT_BUTTON
+      )
+      // make invisible
+      button.click()
+      CreateReport.numberTrainedFormGroup.waitForExist({ reverse: true })
+      // make visible
+      button.click()
+      CreateReport.numberTrainedFormGroup.waitForExist()
+
+      expect(CreateReport.numberTrainedField.getValue()).to.equal(
+        validNumberInput
+      )
+    })
+
+    it("Should not persist previous invalid data when toggling field's visibility", () => {
+      CreateReport.numberTrainedField.setValue(inValidNumberInput)
+      CreateReport.numberTrainedHelpText.waitForExist()
+      // Actually see the validation warning
+      expect(CreateReport.numberTrainedHelpText.getText()).to.equal(
+        "Number trained must be greater than or equal to 1"
+      )
+      const button = CreateReport.getEngagementTypesButtonByName(
+        TESTED_ENGAGEMENT_BUTTON
+      )
+      // make invisible
+      button.click()
+      CreateReport.numberTrainedFormGroup.waitForExist({ reverse: true })
+      // make visible
+      button.click()
+      CreateReport.numberTrainedFormGroup.waitForExist()
+
+      expect(CreateReport.numberTrainedField.getValue()).not.be.equal(
+        inValidNumberInput
+      )
+      expect(CreateReport.numberTrainedField.getValue()).to.be.equal("")
+    })
+
+    it("Should validate visible field", () => {
```

Comment:
I think it would be good to also add a test before this one to show that the validation message is being displayed in the form also.